inductiva/resources/machines.py

```diff
@@ -0,0 +1,112 @@
+              inductiva.resources.create_resource_pool() method.
+        """
+        self.machine_type = machine_type
+        self.num_machines = num_machines
+        self.spot = spot
+        self.disk_size_gb = disk_size_gb
+        self.label = label
+        if machine_group_id is None:
+            self.machine_group_id = resources.create_machine_group_id()
+        else:
+            self.machine_group_id = machine_group_id
+        self.name = self._generate_instance_name()
+
+        self.api_config = api.validate_api_key(inductiva.api_key)
+
+    def start(self):
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            body = InstanceGroup(
+                name=self.name,
+                machine_type=self.machine_type,
+                num_instances=self.num_machines,
+                spot=self.spot,
+                resource_pool_id=self.machine_group_id,
+                disk_size_gb=self.disk_size_gb,
+            )
+            try:
+                logging.info("Creating a machine group. \
+                             This may take a few minutes.")
+                api_instance.create_instance_group(body=body)
+                logging.info("Machine group is successfully created.")
+            except ApiException as e:
+                raise e
+
+    def terminate(self):
+
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            try:
+                logging.info("Terminating machine group. \
+                             This may take a few minutes.")
+                api_instance.delete_instance_group(body=Instance(
+                    name=self.name))
+                logging.info("Machine group is successfully terminated.")
+
+            except ApiException as e:
+                raise e
+
+    def estimate_cost(self):
+        """Returns an estimated cost per hour of a single machine."""
+        with ApiClient(self.api_config) as client:
+            api_instance = InstanceApi(client)
+
+            try:
+                instance_price = api_instance.get_instance_price(body=Instance(
+                    name=self.machine_type))
+                logging.info("Estimated on-demand cost: %f/hour, resulting in \
+                             %f/hour for all machines.",
+                             instance_price.body["on_demand"],
+                             instance_price.body["on_demand"] * \
+                                self.num_machines)
+                logging.info("Estimated spot cost: %f/hour, resulting in \
+                             %f/hour for all machines.",
+                             instance_price.body["preemptible"],
+                             instance_price.body["preemptible"] * \
+                                self.num_machines)
+            except ApiException as e:
+                raise e
+        return instance_price.body
+
+    def _generate_instance_name(self):
```

Comment:
I don't think this should be here in the client. For now, it's OK if it unblocks and allows you to merge this, but this logic of generating unique IDs and stuff like that should be in the backend.

Reply:
yes, we can move it to backend later.